### ISSUE TITLE
Scanning doesn't pick up PSP ISOs compressed into CHDs

### ISSUE THREAD

**Issue by CosmicHeron** [libretro/RetroArch#15966]
### Description
When scanning content (either individually or whole dir) it appears Retroarch does not pick up PSP ISOs that have been compressed into CHDs. If CHDs are imported using a manual scan or manually loaded they can then be launched by the PPSSPP core so this is not a core issue.

This behavior was reproduced using confirmed good ISOs that are picked up by the content scan if left in their uncompressed form. I suspect the database does not have checksums for PSP CHDs?

### Expected behavior
RetroArch is able to pick up PSP ISOs compressed into CHDs as it does for content for other systems (PS1, Saturn, Dreamcast, etc.).

### Actual behavior
RetroArch does not detect PSP ISOs compressed into CHDs.

### Steps to reproduce the bug
1. Get a good PSP ISO and compress it into a CHD file using mame-tools' chdman
2. Load up RetroArch, in the updater get the latest database, optionally install the PPSSPP core
3. Attempt to scan the CHD, note it will not be added to the PSP playlist
4. Scan the ISO, note it will be added to the playlist
5. Optionally load the CHD (either manually or by importing it into the playlist using the manual scan) and launch it using the PPSSPP core, note the game will load and start

### Version
- RetroArch: 1.16.0

### Environment information
- OS: ArchLinux, package installed from the official extra repository

**Comment by i30817** [comment]
This is possible to implement but basically requires special treatment, and probably only using serial scanning.

Although chd has a inbuilt and single match sha256 checksum inbuilt, which you'd think would be able to match games in the database, redump doesn't actually _include_ the chd single checksum except by accident of the dumps using a single file format (in the case of the psp it works, because the psp redump dumps use iso dvds, but in the case of the ps1 for example, you're screwed unless you can code the complicated heuristic to divide tracks and calculate individual track checksums.... and the only example I saw of that didn't work for Dreamcast dumps...)

**Comment by i30817** [comment]
You mentioned that ra picks up chd ps1 games. This is surprising to me actually, because the above. I don't think anything changed with redump, so maybe ra is already picking up checksums in the ps1 but not so in the psp?

**Comment by CosmicHeron** [comment]
So I just re-ran a normal scan on CHDs made from PS1 cue/bins with confirmed redump checksums and RetroArch picked up and added >99% of them to the PS1 playlist. That's why I found it odd it wasn't doing the same for PSP CHDs.

**Comment by i30817** [comment]
Yes, I'm pretty certain it's because the project decided that ps1 games being recognized by the automatic scanner force it to use serials, and that still works with chd.

On the psp on the other hand, it uses checksums, for whatever reason.

**Comment by i30817** [comment]
Actually now that I think about it some more, if RA can search the database for 'any' sha1 checksum, then look up the game that sha1 corresponds, that might be enough to scan chds by checksum.

Chd has a field called 'data sha1' that should suffice to find 99% of the psp chds if the data is in the databases. If it's for a entirely different kind of dump (iso\single track) it doesn't matter until the game has multiple tracks, which none on the psp have.

**Comment by warmenhoven** [comment]
I'm pretty sure this got fixed at some point, there have been a number of improvements and fixes since this was filed. Anyway it works for me.
